src/microphasing.rs

```diff
@@ -243,48 +361,153 @@ pub fn phase_gene<F: io::Read + io::Seek, O: io::Write>(
+                    if offset == exon.end - window_len - current_exon_offset {
+                        read_tree.range((offset - (max_read_len - window_len))..(offset+1)).map(|rec| rec.1).flatten().collect_vec()
+                    }
+                    else {
+                        read_tree.range((offset - (max_read_len - window_len))..(offset - (max_read_len - window_len) + 1)).map(|rec| rec.1).flatten().collect_vec()
+                    }
+                }
+                else {
+                    if offset == exon.start + current_exon_offset {
+                        read_tree.range((offset-(max_read_len - window_len))..(offset+1)).map(|rec| rec.1).flatten().collect_vec()
+                    }
+                    else {
+                        read_tree.range((offset-1)..(offset+1)).map(|rec| rec.1).flatten().collect_vec()
+                    }
+                };
 
+
+
+                debug!("Variants added: {}", added_vars);
+                debug!("Variants deleted: {}", deleted_vars);
                 {
                     // delete rows
-                    observations.cleanup_reads(offset + window_len);
-
+                    if transcript.strand == Strand::Reverse {
+                        observations.cleanup_reads(offset, true);
+                    }
+                    else {
+                        observations.cleanup_reads(offset + window_len, false);
+                    }
                     // delete columns
                     observations.shrink_left(deleted_vars);
 
                     // add new reads
-                    for read in read_buffer.iter() {
-                        if !observations.contains(read)? {
-                            observations.push_read(read.clone(), offset + window_len, offset)?;
+                    debug!("Reads: {}", reads.len());
+                    for read in reads {
+
+                        if transcript.strand == Strand::Reverse {
+                            observations.push_read(read.clone(), offset + window_len, offset, true)?;
+                        }
+                        else {
+                            observations.push_read(read.clone(), offset + window_len, offset, false)?;
                         }
                     }
 
-                    // collect variants
-                    let nvars = variant_buffer.len();
-                    let variants = variant_buffer.iter_mut().skip(
-                        nvars - added_vars
-                    ).map(|rec| Variant::new(rec).unwrap()).flatten().collect_vec();
 
+                    // collect variants
+                    let variants = match transcript.strand {
+                        Strand::Reverse => variant_tree.range_mut(offset..(offset + window_len)).rev()
+                        .map(|var| var.1.clone()).flatten().skip(nvars - added_vars).collect_vec(),
+                        _ => variant_tree.range_mut(offset..(offset + window_len)
```

Comment:
This should rather be `Strand::Forward`.

Reply:
I just saw that it can be unknown as well? I this we should remove the Unknown variant from the Strand enum, and always assume Forward in this case or ignore the transcript and print a warning. This makes the code a bit cleaner.